meta-client/src/main/java/org/apache/helix/metaclient/impl/zk/util/ZkMetaClientUtil.java

```diff
@@ -358,4 +358,25 @@ public static MetaClientException.ReturnCode translateZooKeeperCodeToMetaClientC
         return MetaClientException.ReturnCode.DB_USER_ERROR;
     }
   }
+
+  public static String getZkParentPath(String path) {
+    if (path.equals("/")) {
+      return null;
+    }
+
+    int idx = path.lastIndexOf('/');
+    return idx == 0 ? "/" : path.substring(0, idx);
+  }
+
+  // Splits a path into the paths for each node along the way.
+  // /a/b/c --> /a/b/c, /a/b, /a
+  public static List<String> separateIntoUniqueNodePaths(String path) {
+    ArrayList<String> nodePaths = new ArrayList<>();
```

Comment:
Usually java put interface not concrete class as type. So it should be:

List\<String\> nodePaths = new ArrayList<>();